### PR TITLE
Fix lazy computation of rawCopyrightNotice

### PR DESCRIPTION
**Pull request by keithc-ca**
* the method checks cCopyrightNotice instead of rawCopyrightNotice
* reformatted

Signed-off-by: Keith W. Campbell <keithc@ca.ibm.com>

### CODE REVIEW

File: sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java [hunk 1]
```diff
@@ -67,96 +67,98 @@
...
-	
+
 	public int numberOfPhases() {
 		return phaseList.length;
 	}
-	
+
 	public String phaseName(int phase) {
-		if (phase >= numberOfPhases() )	return "base phase identifier [" + phase + "]";
+		if (phase >= numberOfPhases())
```
Comment (pshipton) [inline sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java]: If you're going to split onto 2 lines, then `{` `}` should be used.

File: sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java [hunk 2]
```diff
@@ -67,96 +67,98 @@
...
 		return phaseList[phase];
 	}
-	
-	public int phaseFromString( String phase ) throws UMABadPhaseNameException {
-		for ( int i=0; i< phaseList.length; i++ ) {
-			if ( phase.equalsIgnoreCase(phaseList[i]) ) return i;
+
+	public int phaseFromString(String phase) throws UMABadPhaseNameException {
+		for (int i = 0; i < phaseList.length; i++) {
+			if (phase.equalsIgnoreCase(phaseList[i]))
```
Comment (pshipton) [inline sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java]: `{` `}`

File: sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java [hunk 3]
```diff
@@ -165,217 +167,229 @@ void processProperties(String path) throws UMAException {
...
+
 	public String getConfigurationName() {
 		return buildSpec.getId();
 	}
-	
+
 	public boolean isValidJcl(String jcl) {
 		for (String validJcl : buildInfo.getJCLs()) {
-			if ( jcl.equals(validJcl)) return true;
+			if (jcl.equals(validJcl))
```
Comment (pshipton) [inline sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java]: `{` `}`

File: sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java [hunk 4]
```diff
@@ -165,217 +167,229 @@ void processProperties(String path) throws UMAException {
...
 		}
-		
-		// hack some special cases in 
-		if ( oldFlag.equals("J9VM_ENV_BUILD_SHARED_LIB_VM")) return "env_buildSharedLibVM";
-		if ( oldFlag.equals("J9VM_ENV_BUILD_STATIC_VM")) return "env_buildStaticVM";
-		if ( oldFlag.equals("J9VM_IVE_JXE_OERELOCATOR")) return "ive_jxeOERelocator";
-		
+
+		// hack some special cases in
+		if (oldFlag.equals("J9VM_ENV_BUILD_SHARED_LIB_VM"))
```
Comment (pshipton) [inline sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java]: `{` `}` for the following 3 ifs

File: sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java [hunk 5]
```diff
@@ -165,217 +167,229 @@ void processProperties(String path) throws UMAException {
...
-		while ( true ) {
+		while (true) {
 			int last = pos;
 			pos = newFlag.indexOf("_", last);
-			if ( pos < 0 ) {
-				if ( last >= newFlag.length() )	break;
-				else pos = newFlag.length();
+			if (pos < 0) {
+				if (last >= newFlag.length())
+					break;
```
Comment (pshipton) [inline sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java]: `{` `}`

File: sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java [hunk 6]
```diff
@@ -165,217 +167,229 @@ void processProperties(String path) throws UMAException {
...
+				else
```
Comment (pshipton) [inline sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java]: `{` `}`

File: sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java [hunk 7]
```diff
@@ -165,217 +167,229 @@ void processProperties(String path) throws UMAException {
...
 			pos += 1;
 		}
 		return flag;
 	}
-	
+
 	public boolean isFlagValid(String flag) {
 		flag = transformFlag(flag);
-		if (flagDefs.getFlagDefinition(flag)==null) return false;
+		if (flagDefs.getFlagDefinition(flag) == null)
```
Comment (pshipton) [inline sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java]: `{` `}`

File: sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java [hunk 8]
```diff
@@ -165,217 +167,229 @@ void processProperties(String path) throws UMAException {
...
-			makefileCopyrightNotice = 
-				"#\n" + 
-				"#     " + getRawCopyrightNotice() +
-				"\n#\n" +
-				"#     This program and the accompanying materials are made available under\n" +
-				"#     the terms of the Eclipse Public License 2.0 which accompanies this\n" +
-				"#     distribution and is available at https://www.eclipse.org/legal/epl-2.0/\n" +
-				"#     or the Apache License, Version 2.0 which accompanies this distribution and\n" +
-				"#     is available at https://www.apache.org/licenses/LICENSE-2.0.\n" +
+			if (majorVersion == null)
```
Comment (pshipton) [inline sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java]: `{` `}`

File: sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java [hunk 9]
```diff
@@ -165,217 +167,229 @@ void processProperties(String path) throws UMAException {
...
+				majorVersion = "0";
+			if (minorVersion == null)
```
Comment (pshipton) [inline sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java]: `{` `}`

File: sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java (hunk 1 above)
Comment (keithc-ca) [reply sourcetools/com.ibm.uma/com/ibm/j9/uma/configuration/ConfigurationImpl.java]: Sure, I'll fix that.

**Comment by pshipton** [comment]
lgtm and the travis build passed. I assume you want to squash before the merge.

**Comment by pshipton** [comment]
Skipping the CI build since it already passed before the squash.

**Comment by DanHeidinga** [comment]
> the method checks cCopyrightNotice instead of rawCopyrightNotice

@keithc-ca Can you expand on the description for this PR?  I see the change to check 
```
+		if (rawCopyrightNotice == null) {
```
instead of 'cCopyrightNotice` but the effect of this is none obvious (at least to me).

**Comment by keithc-ca** [comment]
As it happens, if `cCopyrightNotice` was not null, then `rawCopyrightNotice` would not be null either, but to me *that* was non-obvious. The method that (directly) computes a field should check that field and not some other field. That was not the case for `rawCopyrightNotice`.

**Comment by DanHeidinga** [comment]
> The method that (directly) computes a field should check that field and not some other field.

Or clearly document why the other field is being check.

Thanks for providing the context @keithc-ca!

**Pull request merged**